src/services/faultTreeService.tsx

```diff
@@ -26,14 +26,19 @@ export const findAll = async (): Promise<FaultTree[]> => {
   }
 };
 
-export const find = async (faultTreeUri: string): Promise<FaultTree> => {
+export const find = async (faultTreeUri: string): Promise<{ result: FaultTree; reqProb?: any }> => {
   try {
     const fragment = extractFragment(faultTreeUri);
     const response = await axiosClient.get<FaultTree[]>(`/faultTrees/${fragment}`, {
       headers: authHeaders(),
     });
 
-    return JsonLdUtils.compactAndResolveReferences<FaultTree>(response.data, CONTEXT);
+    let reqProb;
+    if (response.data?.manifestingEvent?.supertypes[0]?.failureRate?.requirement?.upperBound !== undefined) {
```

Comment:
IMO, for readability it would be better to extract `upperBound` with something like:
```
const upperBound = response.data?.manifestingEvent?.supertypes?.[0]?.failureRate?.requirement?.upperBound
if (upperBound !== undefined || upperBound !== null) {
  reqProb = upperBound;
}
```

Reply:
Agree, fixed